lib/sms_autofill.dart

```diff
@@ -203,6 +204,8 @@ class _PhoneFieldHintState extends State<PhoneFieldHint> {
                 ? IconButton(
                     icon: Icon(Icons.phonelink_setup),
                     onPressed: () async {
+                      _focusNode.requestFocus();
```

Comment:
for me this line itself is the fix, and the rest should be removed. Because requesting the focus will trigger the callback and call `_askPhoneHint `

Reply:
You mean `requestFocus()` will be trigger when icon is clicked, so I don't need to call `requestFocus()`, right?

Reply:
No normally if you call `requestFocus ` it should trigger the focus `listener` that will call `_askPhoneHint ` automatically. Here is will be called twice from what I understand

Reply:
Yes, `requestFocus` will trigger focus `listener`, but I set `_hintShown` is `true`, So it will not call `_askPhoneHint()` again.

Reply:
But you set it after requesting the focus so it will to me

Reply:
I think we can remove the focus listener to call `_askPhoneHint()`.
When need to call `_askPhoneHint()`, only icon can trigger it.
What do you think?

Reply:
On my Pixel 4, is `onPressed()` first trigger.
Maybe different device the trigger sequence also different.

Reply:
I want to remove focus listener to call `_askPhoneHint()`.
What do you think?

Reply:
Looks good like this, I prefer keeping the focus listener because the purpose of this field was to not let user enter his phone number manually

Reply:
OK.